answers.md

```diff
@@ -1,19 +1,21 @@
 ### All Answers to Partner Study should be filled out in this file.
 
 1. Single Page Application
-
+    A way to design a web page where you do not need to refresh the browser to see DOM changes.
 2. Compilers
-
+    A program that interprets human readable code to browser or machine readable code or binary.
 3. Bundlers
-
+    Bundlers are programs that re-organize your code files into more efficient or more conventional ways.
 4. Elements
-
+    An element inside of React is typically a piece of html(?) that is returned from components.
 5. Components
-
+    Components are basically classes inside of React. Re-useable pieces of code.
 6. JSX
-
+    A library of sorts that adds functionality to Javascript, such as the ability to define variables as HTML elements.
```

Comment:
JSX isn't a library, it's a JS Syntax eXtension which is specific only to React.

Behind the curtains, React compiles JSX into JavaScript:

```
// JSX
<div className="sidebar" />

// compiled JS
React.createElement(
  'div',
  {className: 'sidebar'},
  null
)
```

[Check out React's docs on JSX](https://reactjs.org/docs/jsx-in-depth.html)